lab/tree_builder.py

```diff
@@ -17,17 +19,45 @@ def __init__(self, root):
         self.root = root
         self.size = 0
 
-    # Function to push to tree 
+    # Function to push to tree
     def push(self, node):
-        # TODO
-        pass
-
-    # Function to remove node from tree 
+        # pointer for the current node
+        currNode = self.root
+        while True:
+            # if value of node to be added is greater, go right
+            if node.val > currNode.val:
+                # if right child exists, go right
+                if currNode.right != None:
```

Comment:
You can change this to say `if currNode.right:` to keep it more pythonic. This boolean will succeed if the object is not `None`